src/notebook.jl

```diff
@@ -173,6 +173,35 @@ macro ingredients(filename)
 	end
 end
 
+# ╔═╡ 36a67c48-f398-4ec1-9f01-b33d647137ee
+"""
+Run a code block using debounce, i.e. only run it when variable list has stabilized through `timeout`.
+
+```julia
+@use_debounce([var], 1) do
+	sleep(2.)
+	@info "stable variable \$var"
+end
+```
+"""
+macro use_debounce(f, deps, timeout)
+	quote
+		ref, set_ref = @use_state(nothing)
+		count_ref = @use_ref(0)
+
+		count_ref.x += 1
+		task = Task(function()
```

Comment:
You can use `@PlutoLinks.use_task` to automatically get an InterruptException if the cell is modified (so that is does not triggers a re-run).

Reply:
I think that I actually want a re-run in this case since my solution simply checks if a "timespan" is kept after `timeout` and tests this out for every value change. I tested the `@use_task` in it and it didn't seem to work.

Reply:
Did you pass the `var` as deps to `@use_task` too ? in the current setting, if you set a debounce of 30s and then edit it to let's 10s after 20s then you would get two re-runs and `@use_task` can prevent that.